### PR TITLE
[HttpKernel] Don't cache controller's reflector inside the request

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #47461
| License       | MIT
| Doc PR        | -

Issue introduced by #46001

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php [hunk 1]
```diff
@@ -40,12 +40,11 @@ public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFa
         $this->argumentValueResolvers = $argumentValueResolvers ?: self::getDefaultArgumentValueResolvers();
     }
 
-    public function getArguments(Request $request, callable $controller): array
+    public function getArguments(Request $request, callable $controller, \ReflectionFunctionAbstract $reflector = null): array
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php:43]: Shouldn't the new argument be added in the interface (in phpdoc, not in the actual signature, for BC reasons) ?

File: src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php [hunk 2]
```diff
@@ -18,25 +18,20 @@
  */
 final class ArgumentMetadataFactory implements ArgumentMetadataFactoryInterface
 {
-    public function createArgumentMetadata(string|object|array $controller, \ReflectionClass $class = null, \ReflectionFunctionAbstract $reflection = null): array
+    public function createArgumentMetadata(string|object|array $controller, \ReflectionFunctionAbstract $reflector = null): array
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php:21]: having no changes in the interface also looks suspicious to me.

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php:43]: works for me, updated

**Review by stof — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**